sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1114,6 +1180,27 @@ def predict(self, X):
         encoded_classes = np.argmax(self.predict_proba(X), axis=1)
         return self.classes_[encoded_classes]
 
+    def staged_predict(self, X):
+        """Predict class at each stage for X.
+
+        This method allows monitoring (i.e. determine error on testing set)
+        after each stage.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+
+        Returns
+        -------
+        y : generator of ndarray of shape (n_samples,)
+            The predicted value of the input samples.
+        """
+        for raw_predictions in self._staged_raw_predict(X):
+            encoded_labels = \
+                np.argmax(self.loss_.predict_proba(raw_predictions), axis=1)
```

Comment:
I think we'd need to have `staged_predict_proba` and `staged_decision_function` too, and we could make use of `staged_predict_proba` here, which itself would make use of `staged_decision_function`